src/store/traction/ont/runs/actions.js

```diff
@@ -0,0 +1,142 @@
+import { handleResponse } from '@/api/ResponseHelper'
+
+const newRun = ({ commit }) => {
+  let run = {
+    id: 'new',
+    instrument_name: null,
+    state: null,
+    flowcell_attributes: [],
+  }
+
+  commit('setCurrentRun', run)
+}
+
+const createRun = async ({ getters, rootGetters }) => {
+  let run = getters.currentRun
+  let request = getters.runRequest
+
+  let instrument_id = getters.instruments.find((i) => i.name == run.instrument_name).id
+
+  let existingPools = rootGetters['traction/ont/pools']
+
+  let flowcell_attributes = run.flowcell_attributes
+    .filter((fc) => fc.flowcell_id && fc.tube_barcode)
+    .map((fc) => {
+      let pool = existingPools.find((p) => p.barcode == fc.tube_barcode)
+      let pool_id = pool ? pool.id : ''
+      return { ...fc, ...{ ont_pool_id: pool_id } }
+    })
+
+  let runPayload = {
+    data: {
+      type: 'runs',
+      attributes: {
+        ont_instrument_id: instrument_id,
+        state: run.state,
+        flowcell_attributes: flowcell_attributes,
+      },
+    },
+  }
+
+  let promise = request.create({ data: runPayload })
+  return await handleResponse(promise)
+}
+
+const updateRun = async ({ getters, rootGetters }) => {
+  let run = getters.currentRun
+  let request = getters.runRequest
+  let instrument_id = getters.instruments.find((i) => i.name == run.instrument_name).id
+
+  let existingPools = rootGetters['traction/ont/pools']
+
+  let flowcell_attributes = run.flowcell_attributes
+    .filter((fc) => fc.flowcell_id && fc.tube_barcode)
+    .map((fc) => {
+      let pool = existingPools.find((p) => p.barcode == fc.tube_barcode)
+      let pool_id = pool ? pool.id : ''
+
+      return { ...fc, ...{ ont_pool_id: pool_id } }
+    })
+
+  let runPayload = {
+    data: {
+      type: 'runs',
+      id: run.id,
+      attributes: {
+        ont_instrument_id: instrument_id,
+        state: run.state,
+        flowcell_attributes: flowcell_attributes,
+      },
+    },
+  }
+
+  let promise = request.update(runPayload)
+  return await handleResponse(promise)
+}
+
+const setInstruments = async ({ commit, getters }) => {
+  let request = getters.instrumentRequest
+  let promise = request.get()
+  const response = await handleResponse(promise)
+  const { success, data: { data } = {}, errors = [] } = response
+
+  if (success && !data.empty) {
```

Comment:
In the setInstruments mutation we can use the populateById method to build an object like this for us

Reply:
@harrietc52